warehouse/core/src/test/java/datawave/ingest/mapreduce/job/SplittableRFileInputFormatTest.java

```diff
@@ -0,0 +1,197 @@
+package datawave.ingest.mapreduce.job;
+
+import static org.apache.accumulo.core.conf.Property.TABLE_CRYPTO_PREFIX;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.accumulo.core.conf.DefaultConfiguration;
+import org.apache.accumulo.core.crypto.CryptoFactoryLoader;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.file.FileSKVIterator;
+import org.apache.accumulo.core.file.FileSKVWriter;
+import org.apache.accumulo.core.file.rfile.RFileOperations;
+import org.apache.accumulo.core.spi.crypto.CryptoEnvironment;
+import org.apache.accumulo.core.spi.crypto.CryptoService;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class SplittableRFileInputFormatTest {
+    private Configuration config;
+
+    @Before
+    public void setup() {
+        Logger.getRootLogger().setLevel(Level.WARN);
+        config = new Configuration();
+    }
+
+    @After
+    public void cleanup() {
+
+    }
+
+    @Test
+    public void testSingleKeySplit() throws IOException, InterruptedException {
+        List<Map.Entry<Key,Value>> data = new ArrayList<>();
+        Key original = new Key(String.format("%08x", 1));
+        data.add(new AbstractMap.SimpleEntry<>(original, new Value()));
+
+        File tmpFile = createRFile(data);
+        try {
+            // get the splits from the file
+            List<InputSplit> splits = SplittableRFileInputFormat.getSplits(config, new Path(tmpFile.toString()));
+
+            // 1 split
+            assertEquals(1, splits.size());
+            RFileSplit rfileSplit = (RFileSplit) splits.get(0);
+            verifySplit(rfileSplit, 0, 2, original, original, 1);
+        } finally {
+            if (tmpFile.exists()) {
+                tmpFile.delete();
+            }
+        }
+    }
+
+    @Test
+    public void testSingleMultiKeySplit() throws IOException, InterruptedException {
+        List<Map.Entry<Key,Value>> data = new ArrayList<>();
+        Key start = new Key(String.format("%08x", 0));
+        Key end = new Key(String.format("%08x", 1));
+        for (int i = 0; i < 2; i++) {
+            Key key = new Key(String.format("%08x", i));
+            data.add(new AbstractMap.SimpleEntry<>(key, new Value(String.format("%08x", i))));
```

Comment:
I couldn't tell where we were using the Value objects. If we aren't, we could just use `new Value()` for all the tests - it'd make the test code a little less dense/easier to read. Along those lines, introducing an `entry(key)` function could slim it up a bunch:

```
data.add(entry(key));
```
Then that `entry()` function could wrap your key in a SimpleEntry with an empty Value.

Reply:
I was originally using the value objects to validate no keys were being lost or duplicated, but once I convinced myself this wasn't happening I ended up dropping those tests. I should bring them back, but also simplify the test creation by abstracting it.